### DISCUSSION TITLE
How to make the company-capf backend use cape-capf-buster? Some candidates are lacking

### DISCUSSION THREAD

**Discussion by IceAsteroid** [company-mode/company-mode#1467] [Q&A]
In haskell-mode, along with the use of ```eglot``` and ```haskell-language-server```, some candidates are lacking in some cases.

For example: Say if I type "import Data." and then the completion panel is popped up, candidates like "List" are lacking. The candidates are exactly 40 ones as the default number of candidates to be shown by the haskell-language-server.

Unless I keep typing until the line "import Data.L", the candidate "List" and other related candidates with "l" letter in them are shown. 

Now I have two options, ask for help to configure a feature like ```cape-capf-buster``` to work to be a more generic solution; Or set the default number of candidates to be a much larger number when such a problem occurs for each language server

Any ideas? :)

**Comment by IceAsteroid** [comment]
For anyone who wants to know, the following configuration of haskell-language-server solved the issue though

```
  (with-eval-after-load 'eglot
    (setq-default eglot-workspace-configuration
		  '((haskell
		     (maxCompletions . 1000))))
    )
```

**Reply by dgutov** [reply]
So... was the issue that the server only returned 40 completions at a time?  In which you, in certain cases, didn't find the desired one.

Then the above is indeed the solution, given that no caches seems to be involved.

It might be possible to do something else in Eglot, but that depends on whether the returned set is marked as "incomplete". Have you tested the same scenario in e.g. VS Code? Does it show more completions?
